lib/autoload/filename_sanitizer.rb

```diff
@@ -0,0 +1,8 @@
+module FilenameSanitizer
```

Comment:
I think you should check the Pathname documentation, I think it can be both a name and a path (a name is a valid path), so I think it might have to be the same function